src/commands/watch/filters.rs

```diff
@@ -137,6 +137,18 @@ fn create_source_globs(cid: &str, c: &v2::Component) -> Option<Vec<String>> {
     }
 }
 
+fn concatenate_glob_friendly(workdir: &str, watch_glob: &str) -> String {
+    // Using Path::join on Windows correctly joins with a backslash. But the watchexec glob
+    // engines hates backslashes. So we must always use forward slashes in glob expressions,
+    // and in workdirs, and for concatenating the two. (We could use Path::join on Unix but
+    // the signature ends up a bit different because to_str returns an Option, so...)
+    if workdir.ends_with('/') {
```

Comment:
What about if `watch_glob` starts with `'/'`?

Reply:
Watching absolute paths is kind of an anti-pattern (and watching absolute paths _with a workdir_ feels even weirder) but okay I guess we can try to address it.  But then we ought to support them on Windows too, which means dealing with `D:\Git\Something\External\**\*.thingy` (or, hopefully, the forward-slashed equivalent) and somehow turning that into a forward-slashable watchexec expression.  I'll cross my fingers and hope that `Path::is_absolute` will magic my problems away!